File: .idea/src/Emailapp/Email.java
package Emailapp;

import java.util.Random;
import java.util.Scanner;

public class Email {

  private String firstname;
  private String lastname;
  private String password;
  private String department;
  private int mailboxcap;
  private int passwordlength = 16;
  private String alteremail;

  //Constructor for First name. Constructor for last name.
  public Email(String firstname, String lastname) {
    this.firstname = firstname;
    this.lastname = lastname;
    System.out.println("Email created for: " + firstname + " " + lastname);


    //call methods asking for Department setting private String setdepartment()and randompassword;
    this.department = setdepartment();
    System.out.println("Department Affiliation: " + this.department);

    this.password = randomPassword(passwordlength);
    System.out.println("Your Password is: " + this.password);
    }

  private String setdepartment(){
    System.out.print("What is the employees department?\nEnter # for Department:\n0 for no specific department\n1 for Development\n2 for Sales\n3 for HR\n4 for Public Safety");
    Scanner in = new Scanner(System.in);

    int Departchoice = in.nextInt();

    if (Departchoice == 1){return "(Development)"; }
    else if (Departchoice == 2){ return "(Sales)"; }
    else if (Departchoice == 3){return "(HR)"; }
    else if (Departchoice == 4){return "(Public Safety)"; }
    else{ return "(None)"; }
    }

  // Constructor for random password.
  private String randomPassword(int length){
     String passwordSet = "abcdefghijklmnopqrstuvwxyz1234567890)(*&^%$#@!";
     char[] password = new char[length];
     for (int i = 0 ; i<length; i++){
          int randm = (int) (Math.random() * passwordSet.length());//import math lang"Math.random"
          password[i] = passwordSet.charAt(randm);
     }
     return new String(password);

  }


 //Constructor for capacity of mailbox

 //Constructor for alternative email.

}

File: .idea/src/EmailGenerator/model/Email.kt
package EmailGenerator.model


const val companySuffix: String = "starkindustries.com"
const val PASSWORD_LENGTH = 16
const val MAILBOX_CAPACITY = 1500

class Email (var firstname: String, var lastname: String){

    var password: String = " "
    var passwordlength: Int = 16
    var department: String = " "
    var email: String = " "
    var alteremail: String = " "

    init {
        // Initialize email and password in the constructor
        setEmployeeDepartment("")
        generateEmployeePassword()
    }

    fun setEmployeeDepartment(department: String) {
        this.department = department
        this.email = "${firstname.lowercase()}.${lastname.lowercase()}@${department.lowercase()}.$companySuffix"
    }

    fun generateEmployeePassword() {
        this.password = createRandomPassword(PASSWORD_LENGTH)
    }

    private fun createRandomPassword(length: Int): String {
        val passwordSet = "abcdefghijklmnopqrstuvwxyz1234567890)(*&^%$#@!"
        val password = CharArray(length)
        for (i in password.indices) {
            val rand = (Math.random() * passwordSet.length).toInt()
            password[i] = passwordSet[rand]
        }
        return String(password)
    }

    fun showInfo(): String {
        return """
            Name: ${firstname.uppercase()} ${lastname.uppercase()}
            Email: $email
            Password: $password
            Mailbox Capacity: $MAILBOX_CAPACITY
        """.trimIndent()
    }
}
